src/hotspot/share/jfr/leakprofiler/sampling/objectSampler.cpp

```diff
@@ -154,11 +155,25 @@ static traceid get_thread_id(JavaThread* thread) {
   if (tl->is_excluded()) {
     return 0;
   }
+  *virtual_thread = tl->is_vthread(thread);
+  return JfrThreadLocal::thread_id(thread);
+}
+
+static JfrBlobHandle get_thread_blob(JavaThread* thread, traceid tid, bool virtual_thread) {
+  assert(thread != NULL, "invariant");
+  JfrThreadLocal* const tl = thread->jfr_thread_local();
+  assert(tl != NULL, "invariant");
+  assert(!tl->is_excluded(), "invariant");
+  if (virtual_thread) {
+    // TODO: blob cache for virtual threads
```

Comment:
Fix this now or after integration?

Reply:
Well spotted. It is an optimization and can happen after integration. I will create an enhancement for tracking; thank you.